HIP/hip-868.md

```diff
@@ -0,0 +1,155 @@
+---
+
+## Abstract
+
+Add support for EIP-6780 SELFDESTRUCT semantics while supporting the Hedera
+account model.
+
+## Motivation
+
+EVM Equivalence requires that Hedera add features supported by Ethereum
+Mainnet (mapped to Hedera semantics). This effort also spans to the rare
+occasion that Ethereum removes features from the EVM.
+
+In this case Ethereum is wanting to remove the SELFDESTRUCT opcode in order to
+make future upgrades achievable. The removal of all contract state as part of
+self-destruction would make future data model changes prohibitively expensive.
+However, there were a number of contracts that depended on side effects of
+SELFDESTRUCT not related to the contract storage and contract code removal,
+namely the transfer of Ether to a beneficiary address. Similarly, some DeFi
+usage patterns used "transient" contracts to store data between calls within the
+same transaction, and self-destruction was key to being a good citizen and not
+leaving garbage around.
+
+To reduce the damage that would occur from simply disabling the opcode Ethereum
+decided to change the behavior of SELFDESTRUCT depending on the age of the
+contract. If the contract existed prior to the start of the transaction the
+operation would change so that neither the code nor the storage was deleted, but
+the ether balance would be sent to the beneficiary. This moved the operation
+into a "sweep" operation that allowed contracts to clean up all balance without
+having to query it. For contracts that did not exist prior to the transaction
+SELFDESTRUCT would continue to operate normally. Balance would be sent to the
+beneficiary, and code and storage would be deleted. Since none of this data had
+been part of the committed data the deletion did not impact any plans for
+changes to their storage model.
+
+Within the EVM Hedera should support a similar model, as contracts will now
+expect SELFDESTRUCT to operate as a sweep in some cases. Outside the EVM,
+however, there is no need to change any semantics. HAPI calls can still be used
+to delete a contract and state expiry can still delete an account.
+
+## Rationale
+
+Hedera will support the [EIP-6780](https://eips.ethereum.org/EIPS/eip-6780)
+change to the SELFDESTRUCT operation. If the contract existed prior to the
+transaction the contract will not be destroyed and the contents will instead be
+swept to the beneficiary. Contracts within the same transaction will be
+destroyed as per prior behavior.
+
+However, since Hedera places "ownership" of token balances with the account
+holding the balance we will need to extend the behavior of the sweep mode to
+also transfer all HTS tokens to the beneficiary. Notably, for a successful
+transaction beneficiary accounts must support the receiving of HTS tokens via
+association configuration. This is consistent with current self-destruct
+behavior and consistent with the spirit of the Ethereum change in that code and
+the account are no longer deleted.
```

Comment:
Currently, if the contract holds HTS token balance the self-destruct fails with `TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES`. This specification is changing the current behavior.